### PR TITLE
Support for Kubernetes vSphere Cloud Provider

### PR DESCRIPTION
**Pull request by abrarshivani**

### CODE REVIEW

File: pkg/model/components/kubelet.go [hunk 1]
```diff
@@ -158,16 +159,8 @@ func (b *KubeletOptionsBuilder) BuildOptions(o interface{}) error {
 	}
 
 	if cloudProvider == fi.CloudProviderVSphere {
-		glog.Info("In kubenet for vsphere cloud provider")
 		clusterSpec.Kubelet.CloudProvider = "vsphere"
 		clusterSpec.Kubelet.HairpinMode = "vmw-bridge"
```
Comment (luomiao) [inline pkg/model/components/kubelet.go]: This hairpin mode seems is also not working for us rt? shall we remove this one too?

File: pkg/apis/kops/componentconfig.go [hunk 2]
```diff
@@ -655,8 +655,11 @@ type CloudConfiguration struct {
 	NodeInstancePrefix *string `json:"nodeInstancePrefix,omitempty"`
 
 	// vSphere cloud-config specs
+	VSphereUsername      *string `json:"vSphereUsername,omitempty"`
```
Comment (luomiao) [inline pkg/apis/kops/componentconfig.go:658]: These fields need to be added into pkg/apis/kops/v1alpha1/componentconfig.go and pkg/apis/kops/v1alpha2/componentconfig.go, and you also need to run conversion-gen to update the auto-generated files.

Comment (abrarshivani) [reply pkg/apis/kops/componentconfig.go:658]: Done

Comment (luomiao) [reply pkg/apis/kops/componentconfig.go:658]: hmm seems you haven't run conversion-gen yet? If you run conversion-gen, a few files like zz_generated.conversion.go should be updated too.

Comment (abrarshivani) [reply pkg/apis/kops/componentconfig.go:658]: I ran ```make codegen``` it didn't generated any files. Do I need to run some other command?

Comment (abrarshivani) [reply pkg/apis/kops/componentconfig.go:658]: Done

**Comment by luomiao** [comment]
LGTM

**Pull request merged**